libs/utils/plot_utils.py

```diff
@@ -0,0 +1,138 @@
+# SPDX-License-Identifier: Apache-2.0
```

Comment:
Is this file copied directly from trappy? If so can you say so in the commit message. If not, if possible, could you copy the file exactly from trappy (with a commit message showing exactly where it came from) then make any changes necessary in a separate commit?

Reply:
If it's going into analysis_module, how would I go about that ? Add a commit that adds the file somewhere, then a commit that squashes that in the file + with the modifications ?

Reply:
Just have a commit like "copy file X from trappy" then a commit that makes the changes. Not sure what you mean by "squash" there...

Reply:
Mm probably not the best choice of words - @derkling suggests to move those utility functions in AnalysisModule, that way the `xlim` thing can be handled directly with `axes.set_xlim(self._trace.x_min, self._trace.x_max)`. This means we don't have to pass `xlim` to `plot_generic()`

Correct me if I'm wrong, but in that case I believe the proper course of action would be:
* Have a commit that says "copy file X from TRAPpy"
* Have another commit that says "move utility methods from file X to AnalysisModule"

Reply:
Ahh I see, OK then the two-commit thing probably isn't worth the hassle.